src/modules/notifications/lib/index.js

```diff
@@ -66,16 +66,19 @@ async function prepareNotification (filter, taskConfig, params, context = {}) {
  * @param {Array} contactData - data from prepare step above
  * @param {String} ref - unique reference for this message batch
  */
-async function sendNotification (queueManager, taskConfig, issuer, contactData, ref) {
+async function sendNotification (queueManager, taskConfig, issuer, contactData, ref, uniqueJobId) {
   // Create event
   const e = eventFactory(issuer, taskConfig, contactData, ref)
   await evt.save(e)
 
   logger.info(`Sending notification reference ${e.referenceCode} ID ${e.eventId}`)
 
   // Schedule messages for sending
-  for (const row of contactData) {
+  for (const [i, row] of contactData.entries()) {
     const n = await notificationFactory(row, taskConfig, e)
+    n.notificationType = taskConfig.task_config_id
+    const rowJobId = `${uniqueJobId}${i}`
+    n.jobId = rowJobId
```

Comment:
Any objections to doing this?

```suggestion
    n.jobId = `${uniqueJobId}${i}`
```

Reply:
<img src="https://media4.giphy.com/media/xT5LMzIK1AdZJ4cYW4/giphy.gif"/>